backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala

```diff
@@ -44,36 +44,39 @@ class UserApi(http: Http, auth: Auth[ApiKey], userService: UserService, xa: Tran
     }
 
   private val changePasswordEndpoint = secureEndpoint.post
-    .in(UserPath / "changepassword")
+    .securityIn(UserPath / "changepassword")
     .in(jsonBody[ChangePassword_IN])
     .out(jsonBody[ChangePassword_OUT])
-    .serverLogic { case (authData, data) =>
-      (for {
-        userId <- auth(authData)
-        _ <- userService.changePassword(userId, data.currentPassword, data.newPassword).transact(xa)
-      } yield ChangePassword_OUT()).toOut
-    }
+    .serverSecurityLogic(authData => auth(authData).toOut)
+    .serverLogic(id =>
+      data =>
+        (for {
+          _ <- userService.changePassword(id, data.currentPassword, data.newPassword).transact(xa)
+        } yield ChangePassword_OUT()).toOut
+    )
 
   private val getUserEndpoint = secureEndpoint.get
     .in(UserPath)
```

Comment:
if `changePasswordEndpoint` uses `.securityIn` for the path, shouldn't we use this here as well?

Reply:
Now I am not sure if `changePasswordEndpoint` require path in `.securityIn` , which way is better in your opinion ?

Reply:
yeah you're right it's fine to keep them all in the regular inputs. The path matching is done on both anyway